src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -2843,7 +2844,7 @@ object Desugar {
     def variable(n: String, s: PScope, context: ExternalTypeInfo): String = name(VARIABLE_PREFIX)(n, s, context)
     def global  (n: String, context: ExternalTypeInfo): String = nameWithoutScope(GLOBAL_PREFIX)(n, context)
     def typ     (n: String, context: ExternalTypeInfo): String = nameWithoutScope(TYPE_PREFIX)(n, context)
-    def field   (n: String, s: StructT): String = nameWithoutScope(s"$FIELD_PREFIX${struct(s)}")(n, s.context)
+    def field   (n: String, @unused s: StructT): String = s"$n$FIELD_PREFIX" // Field names must preserve their equality from the Go level
```

Comment:
Would this pose any problem if two different structs implemented a field with same name but different types?

Reply:
no, the actual field names are only relevant for the encoding of dynamic types.